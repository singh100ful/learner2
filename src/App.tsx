/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  useColorScheme,
} from 'react-native';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import {AppNavigation} from '@src/navigation/AppNavigation';
import {AppProvider} from './provider/AppProvider';

function App(): JSX.Element {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    flex: 1,
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <AppProvider>
      <SafeAreaView style={backgroundStyle}>
        <StatusBar
          barStyle={isDarkMode ? 'light-content' : 'dark-content'}
          backgroundColor={backgroundStyle.backgroundColor}
        />
        <AppNavigation />
      </SafeAreaView>
    </AppProvider>
  );
}

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
});

export default App;
