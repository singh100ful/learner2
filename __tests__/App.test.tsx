/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../src/App';

// Note: import explicitly to use the types shiped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const sleep = (ms: number) =>
  new Promise(resolve => setTimeout(() => resolve(true), ms));

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

it('renders correctly', async () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
  await sleep(10);
});
