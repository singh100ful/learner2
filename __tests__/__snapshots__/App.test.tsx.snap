// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F3F3F3",
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onGestureCancel={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        sheetAllowedDetents="large"
        sheetCornerRadius={-1}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetLargestUndimmedDetent="all"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            collapsable={false}
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "#304FFE",
                    "padding": 15,
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  style={
                    {
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 15,
                      "fontSize": 16,
                      "padding": 10,
                    }
                  }
                  value=""
                />
              </View>
              <RCTScrollView
                ItemSeparatorComponent={[Function]}
                data={null}
                getItem={[Function]}
                getItemCount={[Function]}
                keyExtractor={[Function]}
                onContentSizeChange={[Function]}
                onLayout={[Function]}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={[]}
                style={
                  {
                    "marginHorizontal": 24,
                  }
                }
                viewabilityConfigCallbackPairs={[]}
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </View>
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backTitleVisible={true}
          backgroundColor="#304FFE"
          color="#FFFFFF"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Home"
          titleColor="#FFFFFF"
          topInsetEnabled={false}
          translucent={false}
        />
      </RNSScreen>
    </RNSScreenStack>
  </RNCSafeAreaProvider>
</RCTSafeAreaView>
`;
